packages/desktop/src/renderer/components/widgets/usernameTakenModal/UsernameTakenModal.container.tsx

```diff
@@ -0,0 +1,49 @@
+import { identity, users } from '@quiet/state-manager'
+import React, { useCallback, useEffect } from 'react'
+import { useDispatch, useSelector } from 'react-redux'
+import { useModal } from '../../../containers/hooks'
+import { ModalName } from '../../../sagas/modals/modals.types'
+import CreateUsernameComponent, { UsernameVariant } from '../../CreateUsername/CreateUsernameComponent'
+
+const UsernameTakenModalContainer = () => {
+  const dispatch = useDispatch()
+
+  const isUsernameTaken = useSelector(identity.selectors.usernameTaken)
+  const usernameTakenModal = useModal(ModalName.usernameTakenModal)
+  const registeredUsers = useSelector(users.selectors.certificatesMapping)
+  const user = useSelector(identity.selectors.currentIdentity)
+
+  const registerUsername = useCallback(
+    (nickname: string) => {
+      dispatch(
+        identity.actions.registerUsername({
+          nickname,
+          isUsernameTaken: true,
+        })
+      )
+    },
+    [dispatch]
+  )
+
+  useEffect(() => {
+    if (isUsernameTaken) {
+      usernameTakenModal.handleOpen()
+
+      //IMPORTANT: temporary solution if somehow user recive once wrong information from selector
```

Comment:
Yes, that's true, but I've had weird situations where somehow a user who shouldn't get this modal got this -> the selector returned true once, but after that everything was fine because it returned false, so that's why I had to add this line in else